### PR TITLE
[WKINT-388][Windows][CWS] Add ddprocmon to crash list

### PR DESCRIPTION
**Pull request by derekwbrown**
This change adds ddprocmon to the list of processes that are monitored for crashes by the agent crash detection module; which notifies us if one of our drivers crashes.

Testing Instructions
- Install agent
- Enable _only_ CWS
  - do not enable NPM or USM, because that will invalidate the test
- start the agent
- check the log for the agent crash detection module.  You should see
```
2024-02-22 15:51:08 PST | CORE | DEBUG | (pkg/collector/worker/check_logger.go:44 in CheckStarted) | check:agentcrashdetect | Running check...
2024-02-22 15:51:08 PST | CORE | DEBUG | (pkg/collector/worker/check_logger.go:61 in CheckFinished) | check:agentcrashdetect | Done running check
```
you should NOT see
```
2024-02-22 17:09:41 PST | CORE | INFO | (comp/checks/agentcrashdetect/agentcrashdetectimpl/agentcrashdetect.go:128 in Configure) | Agent Crash Detection module will not run; no required components running
```

For additional testing.  install and load the crash driver (location here). It will crash.  Upon reboot, a crash detection should be generated.  If your host is in `us-1`, then a slack notification should appear in #windows-kernel-intergrations-ops

<!--

### CODE REVIEW

**Comment by derekwbrown** [comment]
[crashdriver.zip](https://github.com/DataDog/datadog-agent/files/14380775/crashdriver.zip)

Github only allows zip files, not `.sys`.
download this zip.  uncompress.  Install with
```
sc.exe create crashdriver type= kernel start= demand binPath= C:\users\Administrator\crashdriver.sys
```
(assumes you unzipped to `c:\users\administrator`).

Then 
start-service crashdriver

your machine will instantly blue screen.

This driver is unsigned.  therefore requires testsigning mode.

File: comp/checks/agentcrashdetect/agentcrashdetectimpl/agentcrashdetect.go [hunk 1]
```diff
@@ -43,14 +43,16 @@ const (
 var (
 	// crashdriver included for testing purposes
 	ddDrivers = map[string]struct{}{
-		"ddnpm":       {},
-		"crashdriver": {},
+		"ddnpm":       {}, // NPM/USM driver, used for network monitoring
+		"ddprocmon":   {}, // process monitoring driver, used for CWS
+		"crashdriver": {}, // this entry exists only for testing purposes.
```
Comment (richl-dd) [inline comp/checks/agentcrashdetect/agentcrashdetectimpl/agentcrashdetect.go:48]: If possible, I think it'd be cleaner if we only included this in the list when we are testing internally. Unfortunately, I'm not such if we have a way to either conditionally compile it in / out or if we could perhaps export ddDrivers and then in test code we could add in to the map(?) Something like that - any ideas?

**Review by richl-dd — APPROVED** [review]
Approved, but heavy emphasis on trying to think of a way to only include `crashdriver` in the map internally during testing.

**Review by brycekahle — APPROVED** [review]

**Pull request merged**